src/Analyzer.PowerShellRuleEngine.UnitTests/PowerShellRuleEngineTests.cs

```diff
@@ -69,32 +82,97 @@ public void AnalyzeTemplate_ValidTemplate_ReturnsExpectedEvaluations(string temp
 
             Assert.AreEqual(expectedErrorCount, failedEvaluations.Count);
 
-            Assert.AreEqual(failedEvaluations.Count, lineNumbers.Length);
-            for (int errorNumber = 0; errorNumber < lineNumbers.Length; errorNumber++)
+            // PSRule evaluations can change order depending on the OS:
+            foreach(var expectedLineNumber in expectedLineNumbers)
             {
-                Assert.AreEqual(lineNumbers[errorNumber], failedEvaluations[errorNumber].Result.LineNumber);
+                var matchingEvaluation = failedEvaluations.Find(evaluation => evaluation.Result.LineNumber == expectedLineNumber);
+                failedEvaluations.Remove(matchingEvaluation);
+            }
+            Assert.IsTrue(failedEvaluations.IsEmptyEnumerable());
+        }
+
+        [DataTestMethod]
+        [DataRow(true, DisplayName = "Repeated rules are excluded when running all the rules")]
+        [DataRow(true, DisplayName = "Repeated rules are excluded when running only the security rules")]
+        public void AnalyzeTemplate_ValidTemplate_ExcludesRepeatedRules(bool runsAllRules)
```

Comment:
I'm not sure what's supposed to be tested here.  The title describes repeated rules being excluded, but the only assertion in the test is whether or not a specific rule appears, which is related to whether or not non-security rules are run, unaffected by repetition.

Reply:
There are two assertions in the code. One checks that whether running all the rules or not an excluded rule is not executed, and the second one validates that when the exclusion baseline is not defined properly the rule is indeed triggering with the test file

Reply:
I see.  Can you add a comments explaining more of how the test works? It would be worth mentioning that the specific id you're looking for is a duplicate of a JSON rule, and should be excluded, unless the baseline file is missing (moved in the test).